drivers/usb/device/usb_dc_dw.c

```diff
@@ -185,36 +177,6 @@ static int usb_dw_reset(void)
 	return 0;
 }
 
-static int usb_dw_clock_enable(void)
-{
-#if defined(CONFIG_SOC_QUARK_SE_C1000)
-	/* 7.2.7 USB Clock Operation */
-	clk_sys_set_mode(CLK_SYS_CRYSTAL_OSC, CLK_SYS_DIV_1);
-
-	/* Enable the USB Clock */
-	QM_SCSS_CCU->ccu_mlayer_ahb_ctl |= QM_CCU_USB_CLK_EN;
-
-	/* Set up the PLL. */
-	QM_USB_PLL_CFG0 = QM_USB_PLL_CFG0_DEFAULT | QM_USB_PLL_PDLD;
-
-	/* Wait for the PLL lock */
-	while (0 == (QM_USB_PLL_CFG0 & QM_USB_PLL_LOCK)) {
-	}
```

Comment:
That, in the other end should be done in driver, via clock_control API, because it might happen on other platforms than quark_se, that it's required to do this also.

The issue with moving too many stuff in soc.c is having driver code scattered in 2 places: most of it in the driver, and some in the soc.c

Reply:
> The issue with moving too many stuff in soc.c is having driver code scattered in 2 places: most of it in the driver, and some in the soc.c

that is exactly my concern. IF this is all way too much to get this driver working somewhere else, you could just leave it as is and ifdef the inclusion of clk.h for systems that use QSMI and then we can remove the SoC related parts from the driver in a more cleaner way

Reply:
ok. I will ifdef clk.h in the usb_dw driver. I will open a issue to add clock control api to enable and disble clock .